blocks/search/graphql-api.js

```diff
@@ -0,0 +1,122 @@
+export const graphQLConfig = {
+  graphQLEndpoint: 'https://search-api-qa-eds.aws.43636.vnonprod.com/search',
+  crossReferenceQueryName: 'rccrossreferencesearch',
+  partNumberQueryName: 'rcpartsearch',
+  filterFacetsQueryName: 'rcfilterfacets',
+  maxProductsPerQuery: 12,
+};
+
+export async function fetchSearchResults({ query, offset, make, model, searchType, category }) {
+  const { graphQLEndpoint, crossReferenceQueryName, partNumberQueryName, maxProductsPerQuery } = graphQLConfig;
+  const queryName = searchType === 'cross' ? crossReferenceQueryName : partNumberQueryName;
+  const graphqlQuery = {
+    query: `
+      query ${queryName}($q: String!, $limit: Int, $offset: Int${searchType === 'parts' ? ', $makeFilter: String, $modelFilter: String' : ''}${category ? ', $categoryFilter: String' : ''}) {
+        ${queryName}(q: $q, limit: $limit, offset: $offset${searchType === 'parts' ? ', makeFilter: $makeFilter, modelFilter: $modelFilter' : ''}${category ? ', categoryFilter: $categoryFilter' : ''}) {
+          count
+          items {
+            score
+            uuid
+            metadata {
+              base_part_number
+              ${searchType === 'cross' ? 'mack_part_number volvo_part_number tenant manufacturer { oem_number name }' : 'model { description make name }'}
+              part_name
+              path
+              part_category
+              image_url
+            }
+          }
+          currentPage
+          numberOfPages
+          facets {
+            key
+            doc_count
+          }
+        }
+      }
+    `,
+    variables: {
+      q: query,
+      limit: maxProductsPerQuery,
+      offset,
+      ...(searchType === 'parts' && { makeFilter: make, modelFilter: model }),
+      ...(category && { categoryFilter: category }),
+    },
+  };
+
+  try {
```

Comment:
Is it worth extracting the try-catch logic into a function?

Reply:
> Is it worth extracting the try-catch logic into a function?

It is, we reduce duplication :) I just did it